src/openms/source/CHEMISTRY/SpectrumAnnotator.cpp

```diff
@@ -0,0 +1,390 @@
+     matched_intensity
+    */
+    defaults_.setValue("list_of_ions_matched", "true", "If set, meta values for matched_ions are added");
+    defaults_.setValidStrings("list_of_ions_matched", ListUtils::create<String>("true,false"));
+    /** adds meta values
+     matched_ions
+    */
+    defaults_.setValue("max_series", "true", "If set, meta values for max_series_type, max_series_size are added");
+    defaults_.setValidStrings("max_series", ListUtils::create<String>("true,false"));
+    /** adds meta values
+     max_series_type
+     max_series_size
+    */
+    defaults_.setValue("S/N_statistics", "true", "If set to 1 isotope peaks of the product ion peaks are added");
+    defaults_.setValidStrings("S/N_statistics", ListUtils::create<String>("true,false"));
+    /** adds meta values
+     sn_by_matched_intensity
+     sn_by_median_intensity
+    */
+    defaults_.setValue("precursor_statistics", "true", "If set, meta values for precursor_in_ms2 are added");
+    defaults_.setValidStrings("precursor_statistics", ListUtils::create<String>("true,false"));
+    /** adds meta values
+     precursor_in_ms2
+    */
+    defaults_.setValue("topNmatch_fragmenterrors", 7, "If set n > 0,  meta values for topN_meanfragmenterror, topN_MSEfragmenterror, topN_stddevfragmenterror are added");
+    /** adds meta values
+     topN_meanfragmenterror
+     topN_MSEfragmenterror
+     topN_stddevfragmenterror
+    */
+    // TODO right now topN matched, but additional information would be gained if the topN would be assembled from NOT ONLY the matched (high intense peak not identified?)
+
+    defaults_.setValue("fragmenterror_statistics", "true", "If set, meta values for median_fragment_error, IQR_fragment_error are added");
+    defaults_.setValidStrings("fragmenterror_statistics", ListUtils::create<String>("true,false"));
+    /** adds meta values
+     median_fragment_error
+     IQR_fragment_error
+    */
+    defaults_.setValue("terminal_series_match_ratio", "true", "If set, meta values for NTermIonCurrentRatio, CTermIonCurrentRatio are added");
+    defaults_.setValidStrings("terminal_series_match_ratio", ListUtils::create<String>("true,false"));
+    /** adds meta values
+     NTermIonCurrentRatio
+     CTermIonCurrentRatio
+    */
+
+    defaultsToParam_();
+  }
+
+  SpectrumAnnotator::SpectrumAnnotator(const SpectrumAnnotator & rhs) :
+    DefaultParamHandler(rhs)
+  {
+  }
+
+  SpectrumAnnotator & SpectrumAnnotator::operator=(const SpectrumAnnotator & rhs)
+  {
+    if (this != &rhs)
+    {
+      DefaultParamHandler::operator=(rhs);
+    }
+    return *this;
+  }
+
+  SpectrumAnnotator::~SpectrumAnnotator()
+  {
+  }
+
+  MSSpectrum<RichPeak1D> SpectrumAnnotator::annotateMatches(const PeptideHit& ph, const MSSpectrum<Peak1D>& spec, const TheoreticalSpectrumGenerator& tg, const SpectrumAlignment& sa) const
+  {
+    MSSpectrum<RichPeak1D> rich_theoretical_spec;
+    MSSpectrum<RichPeak1D> rich_input_spectrum = MSSpectrumHelper::clone(spec);
```

Comment:
instead of the clone function I would prefer a spcialized constructor in MSSpectrum<RichPeak1D>:
template<> MSSpectrum<RichPeak1D>::MSSpectrum(MSSpectrum<Peak1D>){...}